### PR TITLE
Test showing that #1735 was apparently untrue.  Fixes #1735.

### LINKED ISSUE #1735: Record new fleet state when discovering new node via node_metadata_exchange.
If node1 hears, via its REST endpoint, about node37, and it's the most recent node learned, the fleet state is not updated.  This is one of the reasons that we have mismatching fleet states on the monitor.

### PR DESCRIPTION
**Pull request by jMyles**
#1735

### CODE REVIEW

**Comment by cygnusv** [comment]
Needs reading!

File: examples/run_demo_ursula_fleet.py [hunk 1]
```diff
@@ -15,19 +15,6 @@
  along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
 
-"""
```
Comment (KPrasch) [inline examples/run_demo_ursula_fleet.py:18]: Double license in here, eh?

File: nucypher/blockchain/eth/actors.py [hunk 2]
```diff
@@ -117,8 +117,8 @@ def __init__(self, registry: BaseContractRegistry, domains=None, checksum_addres
             self.checksum_address = checksum_address  # type: ChecksumAddress
 
         self.registry = registry
-        if domains:  # StakeHolder config inherits from character config, which has 'domains' - #1580
-            self.network = list(domains)[0]
```
Comment (KPrasch) [inline nucypher/blockchain/eth/actors.py:121]: :sweat_smile: - Good cleanup.

File: nucypher/utilities/seednodes.py [hunk 3]
```diff
@@ -23,54 +23,48 @@
...
+#     except FileNotFoundError:
+#         return dict()   # No static nodes file, No static nodes.
+#     except JSONDecodeError:
+#         raise RuntimeError(f"Static nodes file '{filepath}' contains invalid JSON.")
+#     filtered_static_nodes = {domain: uris for domain, uris in static_nodes.items() if domain in domains}
+#     return filtered_static_nodes
+#
+#
+#
+# def aggregate_seednode_uris(domains: set, highest_priority: Optional[List[str]] = None) -> List[str]:
```
Comment (KPrasch) [inline nucypher/utilities/seednodes.py:51]: I know it's already been merged, but for posterity https://github.com/nucypher/nucypher/issues/2282

**Comment by cygnusv** [comment]
To simplify review, I'm bundling some pending BOM stuff in a single PR, so I'm including this one in #2307 too.

**Pull request closed without merging**